static/src/javascripts/projects/commercial/modules/dfp/add-slot.ts

```diff
@@ -1,17 +1,28 @@
 import type { SizeMapping } from '@guardian/commercial-core';
+import { log } from '@guardian/libs';
 import { Advert } from './Advert';
 import { dfpEnv } from './dfp-env';
 import { enableLazyLoad } from './lazy-load';
 import { loadAdvert } from './load-advert';
 import { queueAdvert } from './queue-advert';
 
-const displayAd = (
-	adSlot: HTMLElement,
-	forceDisplay: boolean,
-	additionalSizes?: SizeMapping,
-) => {
-	const advert = new Advert(adSlot, additionalSizes);
+const createAdvert = (adSlot: HTMLElement, additionalSizes?: SizeMapping) => {
+	try {
+		const advert = new Advert(adSlot, additionalSizes);
+		return advert;
+	} catch {
+		log(
+			'commercial',
+			`Could not create advert. Ad slot: ${
+				adSlot.id
+			}. Additional Sizes: ${JSON.stringify(additionalSizes)}`,
+		);
 
+		return null;
+	}
+};
```

Comment:
Is this logic pretty much the same as that in `fill-advert-slots.ts`? Can we use this `createAdvert` function there?

As an aside, if we reuse this logic somewhere else maybe it should live nearer to the Advert class definition itself.

Reply:
Yep, that's a great idea. I have created a new file `create-advert.ts`